python/lsst/ts/eas/m1m3_thermal.py

```diff
@@ -0,0 +1,208 @@
+#!/usr/bin/env python
+# coding: utf-8
+
+# This file is part of ts_eas.
+#
+# Developed for the Vera C. Rubin Observatory Telescope and Site Systems.
+# This product includes software developed by the LSST Project
+# (https://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+import asyncio
+import logging
+import os
+
+from astropy.time import Time
+from lsst.ts import salobj
+
+
+class ControlLoopManager:
+    def __init__(self, domain: salobj.Domain, log: logging.Logger):
+        """
+        Initializes the ControlLoopManager.
+
+        Parameters
+        ----------
+        domain: salobj.Domain
+            The SALObj domain instance to use.
+
+        log: logging.Logger
+            A logger for log messages.
+        """
+        self.domain = domain
+        self.log = log
+        self.m1m3ts = salobj.Remote(domain, "MTM1M3TS")
+        self.ess = salobj.Remote(domain, "ESS", index=112)
+        self.task = None
+        self.stop_event = asyncio.Event()
+
+        os.environ["LSST_DDS_RESPONSIVENESS_TIMEOUT"] = "15s"
+
+    async def stop_control_loop(self) -> None:
+        """
+        Cancels all running control tasks and waits for them to terminate.
+        """
+        self.log.info("Stopping control loop...")
+        self.stop_event.set()
+        if self.task is not None:
+            self.task.cancel()
+            await asyncio.gather(self.task, return_exceptions=True)
+
+    async def run_control_loop(self, heaterdemand: list[int], fandemand: list[int]):
+        """Runs the control loop for the fans and the heaters.
+
+        Parameters
+        ----------
+        heaterdemand: list[int]
+            The heater power to be applied when the heater is turned on by
+            the MTM1M3TS.heaterFanDemand command. The values range from 0-255,
+            with 255 being 100%.
+        fandemand: list[int]
+            Fan RPM demand for the MTM1M3TS.headerFanDemand command, with
+            255 being 100%.
+        """
+        try:
+            # Wait for remotes to get set up...
+            await asyncio.sleep(5.0)
+
+            mixing = await self.m1m3ts.tel_mixingValve.next()
```

Comment:
Here and everywhere else, when you are grabbing telemetry, make sure you use `flush=True` to ensure you are getting the most up to date value. Also, make sure you add/handle timeouts or this can block forever.